app/models/participant_declaration.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class ParticipantDeclaration < ApplicationRecord
+  include Concerns::ParticipantDeclarationEventRecorder
+  belongs_to :lead_provider
+  belongs_to :early_career_teacher_profile
+
+  scope :for_lead_provider, ->(lead_provider) { where(lead_provider: lead_provider) }
+  scope :active_for_lead_provider, ->(lead_provider) { active.for_lead_provider(lead_provider) }
+  scope :count_active_for_lead_provider, ->(lead_provider) { active_for_lead_provider(lead_provider).count }
+  scope :count_active_for_lead_provider_between, ->(lead_provider, start_date, end_date) { declared_as_between(start_date, end_date).count_active_for_lead_provider(lead_provider) }
```

Comment:
Are we sure we need all of them to be individual scopes? If some are just stepping stones for further scopes, and won't be used outside of the model, I would be tempted to combine them into fewer scopes - at least until we need the intermediate scopes.

Reply:
It makes it easier to double check things in the console and we're going to have different queries. (I actually have used all of them separately in the console to check on different scenarios. As we've not pinned down everything yet, keeping them distinct make modelling and checking edge cases a lot easier.)